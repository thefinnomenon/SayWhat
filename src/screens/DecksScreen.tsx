import React from 'react';
import { Theme, ScreenProps } from '../../types';
import styled from 'styled-components/native';
import { ScrollView } from 'react-native';
import { useNavigation } from 'react-navigation-hooks';
import { useTranslation } from 'react-i18next';
import Deck from '../components/Deck';

export const decks = [
  {
    category: 'Everything',
    imageSrc: require('../../assets/images/everything.jpeg'),
    wordList: require('../../assets/wordlists/everything.ts'),
  },
  {
    category: 'Media',
    imageSrc: require('../../assets/images/entertainment.jpeg'),
    wordList: require('../../assets/wordlists/media.ts'),
  },
  {
    category: 'History',
    imageSrc: require('../../assets/images/history.jpeg'),
    wordList: require('../../assets/wordlists/history.ts'),
  },
  {
    category: 'Sports & Games',
    imageSrc: require('../../assets/images/sports.jpeg'),
    wordList: require('../../assets/wordlists/sports_and_games.ts'),
  },
  {
    category: 'Geography',
    imageSrc: require('../../assets/images/geography.jpeg'),
    wordList: require('../../assets/wordlists/geography.ts'),
  },
  {
    category: 'Around The House',
    imageSrc: require('../../assets/images/house.jpeg'),
    wordList: require('../../assets/wordlists/around_the_house.ts'),
  },
  {
    category: 'Food & Drink',
    imageSrc: require('../../assets/images/food_and_drink.jpeg'),
    wordList: require('../../assets/wordlists/food_and_drink.ts'),
  },
  {
    category: 'Plants & Animals',
    imageSrc: require('../../assets/images/animals_and_plants.jpeg'),
    wordList: require('../../assets/wordlists/animals_and_plants.ts'),
  },
];

export const DecksScreen = () => {
  const { navigate } = useNavigation();
  const { t } = useTranslation();

  return (
    <Container>
      <ScrollView>
        <ContentContainer>
          {decks.map(({ category, imageSrc, wordList }) => (
            <Deck
              key={category}
              category={t(category)}
              imageSrc={imageSrc}
              onPress={() => navigate('Round', { category, wordList })}
            />
          ))}
        </ContentContainer>
      </ScrollView>
    </Container>
  );
};

const Container = styled.View<Theme>`
  flex: 1;
  background: ${props => props.theme.colors.background};
`;

const ContentContainer = styled.View`
  flex: 1;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin: 4px;
`;

DecksScreen.navigationOptions = ({
  screenProps,
}: {
  screenProps: ScreenProps;
}) => ({
  title: screenProps.t('Decks'),
});

export default DecksScreen;
