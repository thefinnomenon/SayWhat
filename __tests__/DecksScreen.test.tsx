import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ThemeManager from '../contexts/ManageThemeContext';
import { I18nextProvider } from 'react-i18next';
import i18 from '../i18n';

import DecksScreen from '../src/screens/DecksScreen';

const renderComponent = () =>
  renderer
    .create(
      <ThemeManager>
        <I18nextProvider i18n={i18}>
          <DecksScreen />
        </I18nextProvider>
      </ThemeManager>,
    )
    .toJSON();

it('renders correctly', () => {
  const tree = renderComponent();
  expect(tree).toMatchSnapshot();
});
