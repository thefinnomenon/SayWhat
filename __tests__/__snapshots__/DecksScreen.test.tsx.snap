// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "flexWrap": "wrap",
              "justifyContent": "center",
              "marginBottom": 8,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 8,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Everything"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/everything.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Everything
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Media"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/entertainment.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Media
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_History"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/history.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              History
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Sports & Games"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/sports.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Sports & Games
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Geography"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/geography.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Geography
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Around The House"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/house.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Around The House
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Food & Drink"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/food_and_drink.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Food & Drink
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderRadius": 20,
              "height": 500,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "width": "45%",
            }
          }
          testID="DECK_Plants & Animals"
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/animals_and_plants.jpeg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "borderRadius": 20,
                  },
                ]
              }
            />
            <BVLinearGradient
              colors={
                Array [
                  0,
                  3003121664,
                ]
              }
              endPoint={
                Object {
                  "x": 0,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0,
                  "y": 0.3,
                }
              }
              style={
                Array [
                  Object {
                    "borderRadius": 20,
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                ]
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 56,
                    "fontWeight": "bold",
                    "marginBottom": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 16,
                  },
                ]
              }
            >
              Plants & Animals
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
